pkg/appmanager/outputConfig.go

```diff
@@ -61,26 +119,105 @@ func (appMgr *Manager) outputConfigLocked() {
+					clientProfCt := virtual.GetProfileCountByContext(customProfileClient)
+					serverProfCt := virtual.GetProfileCountByContext(customProfileServer)
+					if 0 == clientProfCt && 0 == serverProfCt {
+						sslProf := "Common/clientssl"
+						resources[partition].Virtuals[vKey].AddFrontendSslProfileName(sslProf)
+					}
 				}
 			}
 		}
 	}
+
 	for _, profile := range appMgr.customProfiles.profs {
-		resources.CustomProfiles = append(resources.CustomProfiles, profile)
+		initPartitionData(resources, profile.Partition)
+		resources[profile.Partition].CustomProfiles = append(resources[profile.Partition].CustomProfiles, profile)
 	}
 	for _, irule := range appMgr.irulesMap {
-		resources.IRules = append(resources.IRules, *irule)
+		initPartitionData(resources, irule.Partition)
+		resources[irule.Partition].IRules = append(resources[irule.Partition].IRules, *irule)
 	}
 	for _, intDg := range appMgr.intDgMap {
-		resources.InternalDataGroups = append(resources.InternalDataGroups, *intDg)
+		initPartitionData(resources, intDg.Partition)
+		resources[intDg.Partition].InternalDataGroups = append(resources[intDg.Partition].InternalDataGroups, *intDg)
+	}
+
+	// Update resources to conform to the CCCL schema and empty out unneeded fields
+	// so they will be stripped out by the JSON marshaller
+	for partition, partitionConfig := range resources {
+		for i, _ := range partitionConfig.Virtuals {
+
+			resources[partition].Virtuals[i].Enabled = true
+
+			// Add the profiles to the Virtual Server
+			mode := strings.ToLower(resources[partition].Virtuals[i].Mode)
+			if mode == "http" {
+				resources[partition].Virtuals[i].IpProtocol = "tcp"
+				profile := ProfileRef{Partition: "Common", Name: "http", Context: "all"}
+				resources[partition].Virtuals[i].Profiles = append(resources[partition].Virtuals[i].Profiles, profile)
+			} else if mode == "tcp" {
+				resources[partition].Virtuals[i].IpProtocol = "tcp"
+				profile := ProfileRef{Partition: "Common", Name: "tcp", Context: "all"}
+				resources[partition].Virtuals[i].Profiles = append(resources[partition].Virtuals[i].Profiles, profile)
+			} else if mode == "udp" {
+				resources[partition].Virtuals[i].IpProtocol = "udp"
+			}
+
+			// Parse the SSL profile into partition and name
+			if nil != resources[partition].Virtuals[i].SslProfile {
+				// The ssl profile item can be empty or have either
+				// 'f5ProfileName' or 'f5ProfileNames', not both.
+				if "" != resources[partition].Virtuals[i].SslProfile.F5ProfileName {
+					resources[partition].Virtuals[i].Profiles =
+						appendSslProfile(resources[partition].Virtuals[i].Profiles,
+										 resources[partition].Virtuals[i].SslProfile.F5ProfileName,
+										 "clientside")
+                         } else {
+					for _, p := range resources[partition].Virtuals[i].SslProfile.F5ProfileNames {
+						resources[partition].Virtuals[i].Profiles =
+							appendSslProfile(resources[partition].Virtuals[i].Profiles, p, "clientside")
```

Comment:
We have a constant defined for "clientside", so please use customProfileClient instead.